dev/planning-example.py

```diff
@@ -0,0 +1,236 @@
+               description="W.M. Keck 1 Telescope on Mauna Kea, Hawaii")
+
+# It would also be desirable to be able to have a small database of
+# common telescopes.  Maybe this can at first simply take the form of
+# a python module:
+from astroplan import sites
+
+obs = sites.Keck1
+
+# Environmental conditions should be updatable.  Suggest using keyword
+# parameters (this would make it easy to update an internal AltAz coordinate):
+
+obs.set_environment(pressure=0.600 * u.bar, relative_humidity=0.2,
+                    temperature=10 * u.deg_C)
+
+# Dates are assumed to be UTC by default
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
+"""
+
+# Define a time
+from astropy.time import Time
+time_obs = Time(2457189.500000, format='jd')
+
+# sunset
+obs.sunset(date=time_obs, which='nearest') # Default
+obs.sunset(date=time_obs, which='next')
+obs.sunset(date=time_obs, which='previous')
+
+# sunrise
+obs.sunrise(date=time_obs, which='nearest') # Default
+obs.sunrise(date=time_obs, which='next')
+obs.sunrise(date=time_obs, which='previous')
+
+# moon rise
+obs.moon_rise(date=time_obs, which='nearest')
+
+# moon set
+obs.moon_set(date=time_obs, which='nearest')
+
+# The above functions can be called with an `angle` keyword argument to specify
+# a particular horizon angle for rising or setting, or can be called with
+# convenience functions for particular morning/evening twilight.
+# For example, to compute astronomical twilight by specifying the `angle`:
+obs.sunset(date=time_obs, which='next', angle=18*u.degree) 
+
+# evening (astronomical) twilight
+obs.evening_astronomical(date=time_obs)
+
+# evening (nautical) twilight
+obs.evening_nautical(date=time_obs)
+
+# evening (civil) twilight
+obs.evening_civil(date=time_obs)
+
+# morning (nautical) twilight
+obs.morning_nautical(date=time_obs)
+
+# morning (civil) twilight
+obs.morning_civil(date=time_obs)
+
+# morning (astronomical) twilight
+obs.morning_astronomical(date=time_obs)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(date=time_obs)
+
+# what is the moon altitude and azimuth?
+obs.moon_altaz(date=time_obs)
+
+# Other sun-related convenience functions:
+obs.noon(date=time_obs, which='nearest')
+obs.midnight(date=time_obs, which='nearest')
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan import FixedTarget, airmass_plot
+
+# Define a target.
+t1 = FixedTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = FixedTarget(name='Polaris',
```

Comment:
@bmorris3 – OK, I can see how for `FixedTarget`it's appealing to subclass `SkyCoord` to avoid repetitive `target.coord.`. On the other hand, if `target.name` is not the target name, that's also not nice. A mixed option might be to have a `target.coord` member, but then re-expose most or some of it's attributes on the `target` class.

Basically I'm just guessing here what a good design could be ... @eteq, @adrn, @ejeschke ?